### DISCUSSION TITLE
Regex decoding support

### DISCUSSION THREAD

**Discussion by sheepymeh** [microsoft/onnxruntime-genai#485] [New features / APIs]
Is regex decoding or a "JSON mode" available?

**Comment by yufenglee** [comment]
Could you please explain more on this? What is regex decoding?

**Comment by sheepymeh** [comment]
It's decoding that follows a specific format (typically described by a regex). Some examples are:

Outlines: https://github.com/outlines-dev/outlines

SGLang: https://lmsys.org/blog/2024-02-05-compressed-fsm/

This allows the model to generate according to a fixed format that can later be decoded, such as ensuring that output is valid JSON.

**Comment by natke** [comment]
Thank you for the suggestion! I am adding it to our new Features discussions so that we can look into this further

**Comment by kunal-vaishnavi** [comment]
All examples have been updated in [this PR](https://github.com/microsoft/onnxruntime-genai/pull/1939). Please see the READMEs in each example for more information on grammars and tool calling.
